src/main/java/net/snowflake/ingest/utils/Constants.java

```diff
@@ -71,9 +72,48 @@ public class Constants {
   public static final String DROP_CHANNEL_ENDPOINT = "/v1/streaming/channels/drop/";
   public static final String REGISTER_BLOB_ENDPOINT = "/v1/streaming/channels/write/blobs/";
 
-  public static final int PARQUET_MAJOR_VERSION = 1;
   public static final int PARQUET_MINOR_VERSION = 0;
 
+  /**
+   * Iceberg table serialization policy. Use v2 parquet writer for optimized serialization,
+   * otherwise v1.
+   */
+  public enum IcebergSerializationPolicy {
+    NON_ICEBERG,
+    COMPATIBLE,
+    OPTIMIZED;
+
+    public static IcebergSerializationPolicy fromName(String name) {
+      if (name == null) {
+        return NON_ICEBERG;
+      }
+      for (IcebergSerializationPolicy e : IcebergSerializationPolicy.values()) {
+        if (e.name().equalsIgnoreCase(name)) {
+          return e;
+        }
+      }
+      throw new IllegalArgumentException(
+          String.format(
+              "Unsupported ICEBERG_SERIALIZATION_POLICY = '%s', allowed values are %s",
+              name, Arrays.asList(IcebergSerializationPolicy.values())));
+    }
+
+    public ParquetProperties.WriterVersion getParquetWriterVersion() {
+      switch (this) {
+        case NON_ICEBERG:
+        case COMPATIBLE:
+          return ParquetProperties.WriterVersion.PARQUET_1_0;
```

Comment:
Does this mean that non-iceberg tables (which are snowflake managed tables AFAIK) only support parquet v1?

Reply:
The server-side scanner for FDN tables supports Parquet V2. This PR is specific to the Iceberg table feature and does not alter the default behavior for streaming to FDN tables.

Reply:
Thanks!